src/components/tx/ApprovalEditor/index.tsx

```diff
@@ -32,9 +32,9 @@ export const ApprovalEditor = ({
   safeTransaction: SafeTransaction | undefined
   updateTransaction?: (txs: MetaTransactionData[]) => void
 }) => {
-  const [readableApprovals, error, loading] = useApprovalInfos(safeTransaction)
+  const [readableApprovals] = useApprovalInfos(safeTransaction)
 
-  if (readableApprovals?.length === 0 || !safeTransaction) {
+  if (!readableApprovals || readableApprovals?.length === 0 || !safeTransaction) {
```

Comment:
@schmanu this is a regression from when I added the tests but having this early return here also means there can't be an error/loading state. Ideally, we would check if there are any approvals before going into this component or before calling the hook but we would have to "duplicate" the logic from the module. What do you think?

Reply:
If I understand the problem correctly this could help:

Looking at the `useApprovalInfos` hook, the first step is to scan if there are any approvals present, which does not need to be async.
We could consider modifying the `useAsync` result a bit before returning so that it better represents the loading states

i.e.
```ts
export const useApprovalInfos = (safeTransaction: SafeTransaction | undefined) => {
  const [approvals] = useApprovalData(safeTransaction)

  const { balances } = useBalances()

  const hasApprovalSignatures = approvals && approvals.payload && approvals.payload.length > 0

  const [approvalInfos, error, loading] = useAsync<ApprovalInfo[]>(
    async () => {
      if (!hasApprovalSignatures) return Promise.resolve([])

      return Promise.all(
        approvals.payload.map(async (approval) => {
          let tokenInfo: Omit<TokenInfo, 'name' | 'logoUri'> | undefined = balances.items.find(
            (item) => item.tokenInfo.address === approval.tokenAddress,
          )?.tokenInfo
          if (!tokenInfo) {
            tokenInfo = await getERC20TokenInfoOnChain(approval.tokenAddress)
          }

          const amountFormatted = UNLIMITED_APPROVAL_AMOUNT.eq(approval.amount)
            ? PSEUDO_APPROVAL_VALUES.UNLIMITED
            : ethers.utils.formatUnits(approval.amount, tokenInfo?.decimals)

          return { ...approval, tokenInfo: tokenInfo, amountFormatted }
        }),
      )
    },
    [balances.items.length, approvals],
    false, // Do not clear data on balance updates
  )

  return [approvalInfos ?? [], error, hasApprovalSignatures && loading]
}
```

This would only switch to `loading = true` if there are approval signatures present to avoid a flickering approval editor skeleton.

Reply:
Good idea! I've changed the hook so that it always returns `[]` if there are no approvals.